Remove dynamic values from metric names

joreetz-otto opened the pull request:
This will remove values like the broker name or partition number from the metric names. These values are written as label and are currently making access to the metrics harder

Before:

```
topics_test_partitions_0_consumer_lag_stored{client_id="rdkafka",name="rdkafka#producer-1",topics_partitions_broker="3",topics_partitions_fetch_state="none",topics_partitions_leader="3",topics_partitions_partition="0",topics_topic="test",type="producer"} 0
```

After:

```
topics_partitions_consumer_lag_stored{client_id="rdkafka",name="rdkafka#producer-1",topics_partitions_broker="-1",topics_partitions_fetch_state="none",topics_partitions_leader="-1",topics_partitions_partition="-1",topics_topic="test",type="producer"} 0
```

joreetz-otto committed:
Remove dynamic values from metric names
This will remove values like the broker name or partition number from the metric names
These values are written as label and are currently making access to the metrics harder

AndreasBergmeier6176 commented on v0/pkg/prometheus/gen/updater.go:
```diff
@@ -88,18 +87,12 @@ func updateMapped(d *collector.DynamicMap, rlr *label.RecursiveReflector, fv ref
 			delete(keysToDelete, rk)
 			continue
 		}
-		var keyString string
```
I think it would be good to have a test for this.

joreetz-otto replied:
@AndreasBergmeier6176 
Changes are covered in 'v0/pkg/prometheus/gen/collector_test.go' if this is okay for you

joreetz-otto committed:
Edit tests to meet new metric names
 - dynamic parts are not included in metric names anymore
 - we dont need to transform the metric name anymore
